src/data/finset/basic.lean

```diff
@@ -1439,6 +1439,29 @@ begin
   exact λ x, λ hxs hx, h (hx ▸ x.property)
 end
 
+/-- If a `finset` of a subtype is converted to the main type with
+`embedding.subtype`, all elements of the result have the property of
+the subtype. -/
+lemma property_of_mem_map_subtype {p : α → Prop} (s : finset {x // p x}) {a : α}
+    (h : a ∈ s.map (function.embedding.subtype _)) : p a :=
+begin
+  revert h,
+  contrapose,
+  exact not_mem_map_subtype_of_not_property s
+end
```

Comment:
Please swap this lemma with the previous one. For this lemma a proof like `rcases mem_map.1 h with ⟨x, rfl⟩, exact x.2` should work. Then the previous lemma is `mt s.property_of_mem_map_subtype h`.